test/fw/ptl/utils/plugins/ptl_test_data.py

```diff
@@ -149,160 +149,140 @@ def __save_home(self, test, status, err=None):
         else:
             _msg = 'Could not find Server Object in given test object'
             _msg += ', skipping saving post analysis data'
             f.write(_msg + '\n')
             self.logger.warning(_msg)
             f.close()
             return
-        pbs_diag = os.path.join(svr.pbs_conf['PBS_EXEC'],
-                                'unsupported', 'pbs_diag')
+        pbs_snapshot_path = os.path.join(
+            server.pbs_conf["PBS_EXEC"], "sbin", "pbs_snapshot")
+        parent_dir = os.getcwd()
         cur_user = self.du.get_current_user()
-        cmd = [pbs_diag, '-f', '-d', '2']
-        cmd += ['-u', cur_user]
-        cmd += ['-o', pwd.getpwnam(cur_user).pw_dir]
-        if len(svr.jobs) > 0:
-            cmd += ['-j', ','.join(svr.jobs.keys())]
-        ret = self.du.run_cmd(svr_host, cmd, sudo=True, level=logging.DEBUG2)
+        cmd = [
+            pbs_snapshot_path,
+            '--daemon-logs',
+            '2',
+            '--accounting-logs',
+            '2',
+            '--with-sudo']
+        if len(add_hosts) > 0:
+            cmd += ['--additional-hosts=' + ','.join(map(str, add_hosts))]
+        cmd += ['-o', parent_dir]
+        cmd += ['> /dev/null', '2>&1']
+        ret = self.du.run_cmd(server_host, cmd, level=logging.DEBUG2)
         if ret['rc'] != 0:
             _msg = 'Failed to get diag information for '
-            _msg += 'on %s:' % svr_host
+            _msg += 'on %s:' % server_host
             _msg += '\n\n' + '\n'.join(ret['err']) + '\n\n'
             f.write(_msg + '\n')
             self.logger.error(_msg)
             f.close()
             return
         else:
-            diag_re = r"(?P<path>\/.*\/pbs_diag_[\d]+_[\d]+\.tar\.gz).*"
-            m = re.search(diag_re, '\n'.join(ret['out']))
-            if m is not None:
-                diag_out = m.group('path')
-            else:
-                _msg = 'Failed to find generated diag path in below output:'
-                _msg += '\n\n' + '-' * 80 + '\n'
-                _msg += '\n'.join(ret['out']) + '\n'
-                _msg += '-' * 80 + '\n\n'
-                f.write(_msg)
-                self.logger.error(_msg)
+            if len(ret['out']) == 0:
+                self.logger.error('Snapshot command failed')
                 f.close()
                 return
-        diag_out_dest = os.path.join(datadir, os.path.basename(diag_out))
-        if not self.du.is_localhost(svr_host):
-            diag_out_r = svr_host + ':' + diag_out
-        else:
-            diag_out_r = diag_out
-        ret = self.du.run_copy(current_host, diag_out_r, diag_out_dest,
-                               sudo=True, level=logging.DEBUG2)
+
+            snap_out = ret['out'][0].strip(" ")
+            output_tar = snap_out.split(":")[1]
+
+        snap_out_dest = os.path.join(datadir, os.path.basename(snap_out))
+
+        snap_out_r = output_tar.strip(" ")
+        ret = self.du.run_copy(current_host, snap_out_r, snap_out_dest,
```

Comment:
what will happen is server_host is not current_host? can you please try out this scenario?